src/main/java/seedu/address/logic/commands/task/TaskForOneModuleCommand.java

```diff
@@ -0,0 +1,52 @@
+package seedu.address.logic.commands.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_MODULE_CODE;
+
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.CommandType;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.Model;
+
+/**
+ * Lists all tasks in Mod Manager to the user.
+ */
+public class TaskForOneModuleCommand extends TaskCommand {
+
+    public static final String MESSAGE_SUCCESS = "All tasks for the module %s are listed. "
+            + "Returns to the full task list using: task list.";
+    public static final String MESSAGE_USAGE = COMMAND_GROUP_TASK + " " + COMMAND_WORD_MODULE
+            + ": Lists all the tasks for the module in Mod Manager. "
+            + "Parameters: "
+            + PREFIX_MODULE_CODE + " MOD_CODE "
+            + "Example: " + COMMAND_GROUP_TASK + " " + COMMAND_WORD_MODULE + " "
+            + PREFIX_MODULE_CODE + " CS3230 ";
+    public static final String MESSAGE_MODULE_DOES_NOT_EXIST = "The module %1$s does not exist in Mod Manager.";
+
+    private final String moduleCode;
+
+    public TaskForOneModuleCommand(String moduleCode) {
+        this.moduleCode = moduleCode;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws ParseException, CommandException {
+        requireNonNull(model);
+
+        if (!model.hasModuleCode(moduleCode)) {
+            throw new CommandException(String.format(MESSAGE_MODULE_DOES_NOT_EXIST, moduleCode));
+        }
+
+        model.updateFilteredTaskList(task -> task.getModuleCode().equals(moduleCode));
```

Comment:
task.getModuleCode() returns a ModuleCode object while moduleCode is a String, so the list returned will always be empty.

Reply:
Resolved. Thanks!